File: src/1100-Find-K-Length-Substrings-With-No-Repeated-Characters/1100.cpp
class Solution 
{
public:
    int numKLenSubstrNoRepeats(string S, int K) 
    {
        int res = 0, low = -1;
        unordered_map<char, int> usedChar;
        for (int i = 0; i < S.size(); ++i)
        {
            if (usedChar.count(S[i]) and usedChar[S[i]] >= low) low = usedChar[S[i]];
            if (i - low >= K) res++;
            usedChar[S[i]] = i;
        }
        return res;
    }
};

File: src/1100-Find-K-Length-Substrings-With-No-Repeated-Characters/1100.go
func numKLenSubstrNoRepeats(S string, K int) int {
    res, low := 0, -1
    usedChar := make(map[byte]int)
    for i, v := range []byte(S) {
        if uv, ok := usedChar[v]; ok && uv >= low {
            low = uv
        }
        if i - low >= K {
            res++
        }
        usedChar[v] = i
    }
    return res
}